WordPress/gutenberg
# [RNMobile] Hide help button from UBE

Addresses https://github.com/wordpress-mobile/gutenberg-mobile/issues/4339

## Related PRs and Installable Builds

* `Gutenberg Mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4352
* `WPiOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/17644
* `WPAndroid`: https://github.com/wordpress-mobile/WordPress-Android/pull/15674

## Description

As described in https://github.com/wordpress-mobile/gutenberg-mobile/issues/4339, a help button is currently displaying in the bottom right-hand corner of the unsupported block editor (UBE) in both the Android and iOS apps. The button can be seen while logged into WordPress.com. When tapped on, the button isn't working as expected on iOS and prompts users to navigate away from the editor on Android, creating the potential for edits to be lost.

With this PR, the button is hidden from view in order to avoid confusion and potential loss of edits. A high-level overview of the code changes can be found in the `Types of changes` section below.

## How has this been tested?

While logged into a WordPress.com site:

* Using a web browser, add a block that's not supported by Gutenberg Mobile (currently the Markdown block will work).
* Save the draft or publish the post.
* Load the post in the Gutenberg Mobile editor on WPiOS or WPAndroid.
* Choose the option to "Edit using web editor" on the unsupported block.
* Confirm that the help button no longer displays in the bottom right-hand corner of the UBE.

Installable builds have been created for [WPiOS here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17644#issuecomment-988776948) and [WPAndroid here](https://github.com/wordpress-mobile/WordPress-Android/pull/15674#issuecomment-988796513) to help with testing.

## Screenshots 

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/145207216-9f3a7f60-93c6-4aaf-bcd1-0ffcf3d306a6.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/145227468-d21c4c94-7def-4d6a-919c-7efd6d1e855e.png" width="100%"> |

## Types of changes

Bug fix (non-breaking change which fixes an issue), with the following high-level overview of the code changes involved:

### Gutenberg Mobile changes

* A new file, `external-style-overrides.css`, has [been created here](https://github.com/wordpress-mobile/gutenberg-mobile/blob/85070711a4cf92b98c4fc3578a5f2945b2ee6288/resources/unsupported-block-editor/external-style-overrides.css). This new file lives alongside other external/custom scripts [in Gutenberg Mobile's `unsupported-block-editor` folder](https://github.com/wordpress-mobile/gutenberg-mobile/tree/85070711a4cf92b98c4fc3578a5f2945b2ee6288/resources/unsupported-block-editor). Its purpose will be to house CSS that changes parts of the editor that are external to Gutenberg e.g. buttons added by third-party plugins.
* [CSS to hide a help button](https://github.com/wordpress-mobile/gutenberg-mobile/blob/85070711a4cf92b98c4fc3578a5f2945b2ee6288/resources/unsupported-block-editor/external-style-overrides.css#L1) that's added by Jetpack/WP.com (commonly used third-parties amongst app users) has been added to the new CSS file.
* I've dived into more detail around the iOS and Android changes below. For both platforms, I've attempted to apply the same patterns I found used for the existing JS files in Gutenberg Mobile's `unsupported-block-editor` folder, but tweaked for the newly added CSS file.

### Android changes

* A `getOnGutenbergReadyExternalStyles` function that [fetches the contents of the newly created CSS file](https://github.com/wordpress-mobile/WordPress-Android/blob/61ddfd081f1231ba46230197e433c69220ea047f/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/WPGutenbergWebViewActivity.java#L141-L144) has been added to Android's `WPGutenbergWebViewActivity` class.
* Any external styles defined in the new CSS file will now be [processed on the Gutenberg side](https://github.com/WordPress/gutenberg/blob/a2b298a47e8ecb93bf504ca70ddf75c886a24246/packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergWebViewActivity.java#L337-L339) by first calling `getOnGutenbergReadyExternalStyles` and then following the same steps as the other internal CSS files.

### iOS changes

* We previously only referenced `.js` files in Gutenberg Mobile's `unsupported-block-editor` folder. As this PR adds a `.css` file to that folder, it was necessary to add `css` to [the global pattern here](https://github.com/wordpress-mobile/gutenberg-mobile/blob/bcb75bd442f453716914c59427896ef7f1a058f8/Gutenberg.podspec#L21) in the Gutenberg Mobile repository
* In the iOS repository's [GutenbergWebViewController.swift file](https://github.com/wordpress-mobile/WordPress-iOS/blob/a09fbb771a0d5e78322172bbe94139352a09f9fe/WordPress/Classes/ViewRelated/Gutenberg/GutenbergWeb/GutenbergWebViewController.swift#L109), a `onGutenbergReadyStyles` function calls `loadCustomStyles()`, which then [fetches the contents of the newly created CSS file](https://github.com/wordpress-mobile/WordPress-iOS/blob/a09fbb771a0d5e78322172bbe94139352a09f9fe/WordPress/Classes/ViewRelated/Gutenberg/GutenbergWeb/GutenbergWebViewController.swift#L109). This follows the same pattern as, [loadCustomScript()](https://github.com/wordpress-mobile/WordPress-iOS/blob/a09fbb771a0d5e78322172bbe94139352a09f9fe/WordPress/Classes/ViewRelated/Gutenberg/GutenbergWeb/GutenbergWebViewController.swift#L100), the function used to process `.js` files.
* In order to replicate the way other CSS files are processed in the Gutenberg repository, it was necessary to make [`getContent()`](https://github.com/WordPress/gutenberg/blob/95929b58e4f3a6bdc3f2df3cb612858d5051952b/packages/react-native-bridge/ios/SourceFile.swift#L23)  and [`toJsScript`](https://github.com/WordPress/gutenberg/blob/95929b58e4f3a6bdc3f2df3cb612858d5051952b/packages/react-native-bridge/ios/GutenbergWebFallback/FallbackJavascriptInjection.swift#L71) public for use in the new `loadCustomStyles()` function.
* Finally, `onGutenbergReadyStyles()` is called in all the same places as `injectEditorCssScript()` in [the GutenbergWebSingleBlockViewController.swift file](https://github.com/WordPress/gutenberg/blob/95929b58e4f3a6bdc3f2df3cb612858d5051952b/packages/react-native-bridge/ios/GutenbergWebFallback/GutenbergWebSingleBlockViewController.swift#L170). 

## Checklist:
